### PR TITLE
feat: (Core|Platform) migrate to 0.16.1 styles

### PR DESCRIPTION
**Pull request by JKMarkowski**
#### Please provide a link to the associated issue.
part of https://github.com/SAP/fundamental-ngx/issues/3684
#### Please provide a brief summary of this pull request.
Changes done in this PR are described in [here](https://github.com/SAP/fundamental-ngx/issues/3684#issuecomment-782057218)

This is PR that applies changes due to migration to newest `0.16.1` version of styles. Shortly what's added:
- A lot of removal of unnecessary scss code
- List enhancements
- Some minor features / changes to code

#### **Breaking Changes**:
 - It's required to add `fd-list-title` inside `fd-list-group-header`.
 - List with busy indicator has been changed - now there is `[growing]` input and `fd-list-title-text` directive

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: apps/docs/src/app/core/component-docs/list/list-docs.component.ts [hunk 1]
```diff
@@ -162,6 +163,14 @@ export class ListDocsComponent {
         }
     ];
 
+    listInteractive: ExampleFile[] = [
+        {
+            language: 'html',
+            code: listInteractiveHtml,
+            fileName: 'list-interactive-example'
```
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/list/list-docs.component.ts:170]: You are missing `component`

**Comment by mikerodonnell89** [comment]
Should the first column here have its top aligned with the second column? Not seeing a mention of this in the specs, but how it looks here feels awkward to me

Screen Shot 2021-02-26 at 1 30 13 PM

**Comment by mikerodonnell89** [comment]
Breaking changes say

> * One **Breaking Change** - It's required to add `fd-list-title` inside `fd-list-header`.

but it's actually `fd-list-group-header`

**Review by InnaAtanasova — COMMENTED** [review]
Looks like noVerticalGap and noHorizontalGap are not applied when used individually:
Screen Shot 2021-02-26 at 3 37 48 PM

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
Looks pretty good! Just a few things, see comments

File: apps/docs/src/app/core/component-docs/list/list-docs.component.ts (hunk 1 above)
Comment (JKMarkowski) [reply apps/docs/src/app/core/component-docs/list/list-docs.component.ts:170]: It's not needed for `only-html` examples

**Comment by JKMarkowski** [comment]
> Should the first column here have its top aligned with the second column? Not seeing a mention of this in the specs, but how it looks here feels awkward to me
> 
> Screen Shot 2021-02-26 at 1 30 13 PM

@mikerodonnell89 
Left column has been changed to be always on top - it's up to example, because on styles we have only left with more than 1 column.

**Comment by JKMarkowski** [comment]
@InnaAtanasova,@mikerodonnell89 Thanks for review - your comments have been addressed

File: apps/docs/src/app/core/component-docs/list-byline/examples/list-byline-interactive-example/list-byline-interactive-example.component.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<ul fd-list [byline]="true">
+    <li fd-list-item [interactive]="true">
+        <span fd-list-thumbnail><fd-icon glyph="activate"></fd-icon></span>
+        <div fd-list-content>
+            <div fd-list-title>Title</div>
+            <div fd-list-byline>Byline Text</div>
+        </div>
+    </li>
+    <li fd-list-item [interactive]="true">
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply apps/docs/src/app/core/component-docs/list-byline/examples/list-byline-interactive-example/list-byline-interactive-example.component.html:9]: Added selected state and interactive + selected

File: apps/docs/src/app/core/component-docs/layout-grid/examples/layout-grid-no-gap-example.component.html [hunk 3]
```diff
@@ -23,3 +23,61 @@
...
+
+    <div fdLayoutGridCol="8">
+        <example-layout-grid-block>Size: 8</example-layout-grid-block>
+    </div>
+</fd-layout-grid>
+
+
+<br><br><br>
+
+<fd-layout-grid [noVerticalGap]="true">
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply apps/docs/src/app/core/component-docs/layout-grid/examples/layout-grid-no-gap-example.component.html:58]: Removed old scss code and added import from fundamental-styles

**Review by InnaAtanasova — APPROVED** [review]

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Review by mikerodonnell89 — APPROVED** [review]
LGTM!

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Review by KevinOkamoto — APPROVED** [review]

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Pull request merged**